superset-frontend/src/components/Select/Select.tsx

```diff
@@ -361,13 +376,20 @@ const Select = ({
   };
 
   useEffect(() => {
-    const foundOption = hasOption(searchedValue, selectOptions);
-    if (isAsync && !foundOption) {
+    const allowFetch = !fetchOnlyOnSearch || searchedValue;
+    if (isAsync && loadingEnabled && allowFetch) {
```

Comment:
Do we really need to check whether `loadingEnabled` is `true` here?

Reply:
Yes. Because no query should be fired before the user opens the select for the first time.